src/common/lib/lnurl.ts

```diff
@@ -14,9 +14,10 @@ import { bech32Decode } from "../utils/helpers";
 
 const fromInternetIdentifier = (address: string) => {
   // email regex: https://emailregex.com/
+  // modified to allow _ in subdomains
   if (
     address.match(
-      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-_0-9]+\.)+[a-zA-Z]{2,}))$/
```

Comment:
Is a `_` in the spec for domains? I thought that's only letters, digits, and hyphen?

Reply:
It's allowed in subdomains, I think it may not be allowed in second level domains though 🤷‍♂️

Reply:
that regex is too complicated :) 
<img width="1220" alt="image" src="https://user-images.githubusercontent.com/318/206028420-f6a8d5b5-02fd-4d6f-9177-1864b12a2552.png">

we now match also `_` in the domain. but that's fine :)
thanks for the PR.